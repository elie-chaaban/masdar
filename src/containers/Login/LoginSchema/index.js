import * as Yup from 'yup';

const loginSchema = Yup.object().shape({
  userName: Yup.string().min(2, 'Too Short!').max(50, 'Too Long!').required('Required'),
  password: Yup.string().min(2, 'Too Short!').max(50, 'Too Long!').required('Required')
});

export default loginSchema;
