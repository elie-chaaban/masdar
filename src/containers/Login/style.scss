@import '../../assets/styles/fonts';
body {
  background-image: url('../../assets/images/login-background.png');
  background-repeat: no-repeat;
  background-position: bottom;
  background-size: cover;
  overflow: hidden;
}

.login-form {
  width: 30%;
  position: absolute;
  top: 40%;
  padding: 4rem;
  left: 50%;
  transform: translate(-50%, -50%);
}
input.form-control {
  margin-top: 20px;
  background-color: transparent !important;
  padding: 10px 40px;
  border-radius: 20px;
  border: none;
  position: relative;
  z-index: 12;
  height: 30px;
}
.form-group {
  position: relative;
  width: 90%;
  margin: 0 auto !important;
  .form-control::placeholder {
    color: black;
  }
}
.form-group::after {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background: linear-gradient(to right, #5271ad, #a064a3);
  content: '';
  z-index: 1;
  width: 102%;
  height: 115%;
  border-radius: 50px;
}
.login-btn {
  margin-top: 30px !important;
  border-radius: 14px !important;
  font-weight: bold;
  font-weight: bold;
  color: white;
  background-color: #4eaba9 !important;
  width: 290px;
  height: 57px;
}
.invalid-feedback {
  font-size: 100% !important;
}
@media screen and (max-width: 768px) {
  .login-form {
    margin-top: 20px;
    width: auto;
    max-width: 100%;
    height: 15%;
    padding: 0px;
  }

  .section1 {
    padding: 5px 15px 0px 45px;
    width: auto;
    height: 100%;
  }
}
.expire-modal {
  width: 100vw;
  height: 100vh;
  justify-content: end;
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgba($color: #000, $alpha: 0.5);
  z-index: 1000;
  .m-content {
    width: 15%;
    margin-top: 20px;
    height: 10%;
    padding: 20px;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    background: linear-gradient(to right, #5271ad, #a064a3);
    border-radius: 5px;
    box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15) !important;
    color: white;
    text-align: center;
    h3,
    p {
      width: 100%;
    }
  }
}
