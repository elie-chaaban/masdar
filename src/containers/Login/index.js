import React, {useState, Suspense, useEffect} from 'react';
import {Formik, Form} from 'formik';
import {useDispatch, useSelector} from 'react-redux';
import {authenticateUser, toggleExpireModal} from '../../reduxStore/actions';
import {LoaderFull} from '../../components/Utils';
import {errorNotification} from '../../components/Utils/Notifications';
import {useAuthContext} from '@asgardeo/auth-react';
import {isMobile} from 'react-device-detect';

import './style.scss';

const Login = () => {
  const {signIn} = useAuthContext();
  const [loading, setLoading] = useState(false);
  const dispatch = useDispatch();
  const {isExpireModalOpen} = useSelector((s) => s.user);

  const login = async () => {
    try {
      setLoading(true);
      await signIn();
      dispatch(authenticateUser());
    } catch (error) {
      console.log(error);
      errorNotification(error);
    } finally {
      setLoading(false);
    }
  };
  useEffect(() => {
    if (isExpireModalOpen) {
      var timeout = setTimeout(() => {
        console.log('run');
        dispatch(toggleExpireModal());
      }, 3000);
    }
    return () => {
      if (isExpireModalOpen) clearTimeout(timeout);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <Suspense fallback={<LoaderFull />}>
      <Formik initialValues={{}} onSubmit={() => login()}>
        {({errors, touched}) => (
          <Form className="login-form shadow-lg ">
            {/* <img src={logo} alt="logo" className="img-fluid d-block mx-auto logo" />   */}
            <button
              disabled={loading}
              className="btn px-5 text-white d-block mx-auto login-btn"
              name="Login"
              type="submit"
              style={isMobile ? {fontSize: 20} : {}}
            >
              {loading ? 'Logging In..' : 'Login'}
            </button>
          </Form>
        )}
      </Formik>
    </Suspense>
  );
};
export default Login;
