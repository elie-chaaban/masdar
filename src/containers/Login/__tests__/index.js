import React from 'react';
import {Provider} from 'react-redux';
import Login from '..';
import {mockStore} from '../../../testUtils';
import {mount} from 'enzyme';
import {act} from 'react-dom/test-utils';

jest.mock('../../../components/Utils/Notifications', () => ({
  ...jest.requireActual('../../../components/Utils/Notifications'),
  successNotification: jest.fn(),
  errorNotification: jest.fn()
}));

// let resolve;
// jest.mock('../../../services/auth', () => ({
//   ...jest.requireActual('../../../services/auth'),
//   loginUser: jest.fn(
//     () =>
//       new Promise((res) => {
//         resolve = res;
//       })
//   )
// }));
// jest.mock('../../../reduxStore/authReducer/actions.js', () => ({
//   __esModule: true,
//   default: () => ({
//     type: 'AUTHENTICATE_USER'
//   })
// }));
// test('Login Page without expire', async () => {
//   let dom, username, password;
//   const store = mockStore({user: {isExpireModalOpen: false}});
//   await act(async () => {
//     dom = mount(
//       <Provider store={store}>
//         <Login />
//       </Provider>
//     );
//   });
//   username = dom.find("input[name='userName']");
//   password = dom.find("input[type='password']");
//   expect(dom.find('.text-warning')).toHaveLength(0);
//   expect(dom).toMatchSnapshot();
//   expect(dom.find('form.login-form')).toHaveLength(1);
//   const submit = () => dom.find("button[type='submit']").simulate('submit');
//   await act(async () => {
//     submit();
//   });
//   dom.update();
//   let errors = dom.find('.invalid-feedback');
//   errors.forEach(function (item) {
//     expect(item.text()).toBe('Required');
//   });
//   await act(async () => {
//     await username.simulate('change', {
//       target: {value: 'ad', name: 'userName'}
//     });
//     await password.simulate('change', {
//       target: {value: 'ad', name: 'password'}
//     });
//     submit();
//   });
//   dom.update();
//   errors = dom.find('.invalid-feedback');
//   errors.forEach(function (item) {
//     expect(item.text()).toBe('Too Short!');
//   });
//   await act(async () => {
//     await username.simulate('change', {
//       target: {value: 'adelc', name: 'userName'}
//     });
//     await password.simulate('change', {
//       target: {value: '12345678', name: 'password'}
//     });
//   });
//   dom.update();
//   errors = dom.find('.invalid-feedback');
//   expect(errors).toHaveLength(0);
//   await act(async () => {
//     submit();
//   });
//   dom.update();
//   expect(dom.find("[type='submit']").text()).toBe('login In! Please wait...');
//   await act(async () => {
//     resolve(true);
//   });
//   dom.update();
//   expect(dom.find("[type='submit']").text()).toBe('Login');
//   expect(store.getActions()).toHaveLength(1);
//   expect(store.getActions()[0]).toEqual({type: 'AUTHENTICATE_USER'});
//   jest.clearAllMocks();
//   dom.unmount();
// });
test('Login Page with expire', async () => {
  let dom;
  const store = mockStore({user: {isExpireModalOpen: true}});
  await act(async () => {
    dom = mount(
      <Provider store={store}>
        <Login />
      </Provider>
    );
  });
  expect(dom).toMatchSnapshot();
  // expect(dom.find('.text-warning')).toHaveLength(1);
  jest.clearAllMocks();
  dom.unmount();
});
