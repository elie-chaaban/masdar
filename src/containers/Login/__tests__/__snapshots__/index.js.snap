// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page with expire 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Login>
    <Suspense
      fallback={<LoaderFull />}
    >
      <Formik
        initialValues={
          Object {
            "password": "",
            "userName": "",
          }
        }
        onSubmit={[Function]}
      >
        <Form
          className=" login-form shadow-lg "
        >
          <form
            action="#"
            className=" login-form shadow-lg "
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <img
              alt="logo"
              className="img-fluid d-block mx-auto logo"
              src="logo.png"
            />
            <button
              className="btn login-btn px-5 text-white d-block mx-auto"
              disabled={false}
              name="Login"
              type="submit"
            >
              Login
            </button>
          </form>
        </Form>
      </Formik>
    </Suspense>
  </Login>
</Provider>
`;